### PR TITLE
feat(lua): vim.region accepts getpos() arg

### PR DESCRIPTION
**Pull request by uga-rosa**

### CODE REVIEW

File: runtime/lua/vim/_editor.lua [hunk 1]
```diff
@@ -400,8 +400,8 @@ end
 --- Get a table of lines with start, end columns for a region marked by two points
 ---
 ---@param bufnr integer number of buffer
----@param pos1 integer[] (line, column) tuple marking beginning of region
----@param pos2 integer[] (line, column) tuple marking end of region
+---@param pos1 integer[]|string (line, column) tuple marking beginning of region or special characters same as getpos()
```
Comment (justinmk) [inline runtime/lua/vim/_editor.lua:403]:
```suggestion
---@param pos1 integer[]|string start of region as a (line, column) tuple or string accepted by |getpos()|
```

**Comment by uga-rosa** [comment]
Do I need to update news.txt?

**Comment by dundargoc** [comment]
@clason

**Comment by uga-rosa** [comment]
@clason

**Comment by clason** [comment]
Yes, a short `news.txt` entry is appropriate.

But it would be good to adapt `vim.highlight` to these changes, since `on_yank` currently uses `getpos` itself; the strings can now be simply passed down to `vim.region` (via `vim.highlight.range`).

**Review by clason — APPROVED** [review]

**Comment by clason** [comment]
Tests fail now; I think the inputs have to be validated better.

File: runtime/doc/news.txt [hunk 2]
```diff
@@ -25,6 +25,8 @@ The following new APIs or features were added.
 
 • ...
```
Comment (zeertzjq) [inline runtime/doc/news.txt:27]:
```suggestion
```

**Comment by uga-rosa** [comment]
Organized commits.

**Review by clason — APPROVED** [review]

**Pull request merged**